File: WpfSamples/XBAPHostingSilverlightSample/vb/WpfHostingSilverlight/ScriptableClass.vb
Imports System.ComponentModel ' INotifyPropertyChanged
Imports System.Runtime.InteropServices ' ComVisibleAttribute
Imports System.Security 'Security Permissions
Imports System.Security.Permissions 'Security Permissions

Namespace WPFBrowserApplication
	<PermissionSet(SecurityAction.Assert, Name:="FullTrust"), ComVisible(True)> _
	Public Class ScriptableClass
		' Method that can be called from HTML script 
		Public Sub SendMessageToWPF(ByVal msg As String)
			MessageBox.Show("Your WPF application has received a message: " & msg)
		End Sub


	End Class
End Namespace

File: WpfSamples/XBAPHostingSilverlightSample/cs/WpfHostingSilverlight/ScriptableClass.cs
using System.ComponentModel; // INotifyPropertyChanged
using System.Windows; // MessageBox
using System.Runtime.InteropServices; // ComVisibleAttribute
using System.Security; //Security Permissions
using System.Security.Permissions; //Security Permissions

namespace WPFBrowserApplication
{
    [PermissionSet(SecurityAction.Assert, Name="FullTrust")]
    [ComVisible(true)]
    public class ScriptableClass
    {
        // Method that can be called from HTML script 
        public void SendMessageToWPF(string msg)
        {
            MessageBox.Show("Your WPF application has received a message: " + msg);
        }

       
    }
}
